lib/ex_plasma/output.ex

```diff
@@ -0,0 +1,197 @@
+  ...> 50, 125, 202, 87, 133, 226, 40, 180, 148, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+  ...> 0, 0, 0, 0, 0, 0, 0, 0, 136, 13, 224, 182, 179, 167, 100, 0, 0>>
+  iex> ExPlasma.Output.decode(encoded)
+  %ExPlasma.Output{
+    output_data: %{
+      amount: 1000000000000000000,
+      output_guard: <<205, 193, 229, 59, 220, 116, 187, 245, 181, 247, 21, 214, 50, 125, 202, 87, 133, 226, 40, 180>>,
+      token: <<0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0>>},
+    output_id: nil,
+    output_type: 1
+  }
+  """
+  @spec decode(rlp()) :: t()
+  def decode(data) when is_list(data), do: do_decode(data)
+  def decode(data), do: data |> ExRLP.decode() |> do_decode()
+
+  @doc """
+
+  ## Examples
+
+  # Encode as an Output
+
+  iex> output = %ExPlasma.Output{
+  ...>   output_data: %{
+  ...>     amount: 1000000000000000000,
+  ...>     output_guard: <<205, 193, 229, 59, 220, 116, 187, 245, 181, 247, 21, 214, 50, 125, 202, 87, 133, 226, 40, 180>>,
+  ...>     token: <<0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0>>},
+  ...>   output_id: nil,
+  ...>   output_type: 1
+  ...> }
+  iex> ExPlasma.Output.encode(output)
+  <<245, 1, 243, 148, 205, 193, 229, 59, 220, 116, 187, 245, 181, 247, 21, 214,
+  50, 125, 202, 87, 133, 226, 40, 180, 148, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+  0, 0, 0, 0, 0, 0, 0, 0, 136, 13, 224, 182, 179, 167, 100, 0, 0>>
+
+  # Encode as an Input
+
+  iex> output = %ExPlasma.Output{
+  ...>   output_data: nil,
+  ...>   output_id: %{
+  ...>     blknum: 1,
+  ...>     oindex: 0,
+  ...>     position: 1000000000,
+  ...>     txindex: 0
+  ...>   },
+  ...>   output_type: nil
+  ...> }
+  iex> ExPlasma.Output.encode(output, as: :input)
+  <<0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 59, 154, 202, 0>>
+  """
+  def encode(%{} = output, as: :input), do: to_rlp_id(output)
+  def encode(%{} = output), do: output |> to_rlp() |> ExRLP.encode()
+
+  @doc """
+
+  ## Example
+
+  iex> encoded_position = << 59, 154, 202, 0>>
+  iex> ExPlasma.Output.decode_id(encoded_position)
+  %ExPlasma.Output{
+    output_data: nil,
+    output_id: %{
+      blknum: 1,
+      oindex: 0,
+      position: 1000000000,
+      txindex: 0
+    },
+    output_type: nil
+  }
+  """
+  def decode_id(data), do: data |> :binary.decode_unsigned(:big) |> do_decode()
+
+  @doc """
+  Encode an Output into RLP bytes
+
+  ## Example
+
+  iex> output = %{
+  ...>      output_id: nil,
+  ...>      output_type: 1,
+  ...>      output_data: %{output_guard: <<1::160>>, token: <<0::160>>, amount: 1}
+  ...>    }
+  iex> ExPlasma.Output.to_rlp(output)
+  [<<1>>, [<<0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1>>, <<0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0>>, <<1>>]]
+  """
+  @spec to_rlp(t()) :: binary()
+  def to_rlp(%{output_type: nil}), do: nil
+  def to_rlp(%{output_type: type} = output), do: output |> get_output_type(type).to_rlp()
```

Comment:
```suggestion
  def to_rlp(%{output_type: type} = output), do: get_output_type(output, type).to_rlp()
```

> We at the Elixir core team consider a smell to use the pipe just once: value |> Some.fun. The pipe was designed to express pipelines and you don’t have a pipeline if you have only one call. :slightly_smiling_face: Therefore it would be nice to have a check that guarantees you never pipe only once.